### PR TITLE
cache instances by port in push context

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/27117

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
good, i was planning to do this.

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -625,6 +625,19 @@ func GetServiceAccounts(svc *Service, ports []int, discovery ServiceDiscovery) [
 	return sa.UnsortedList()
 }
 
+// GetServiceAccountsFromInstances returns service accounts for a service derived from passed in instances.
+func GetServiceAccountsFromInstances(svc *Service, instances []*ServiceInstance) []string {
```
Comment (howardjohn) [inline pilot/pkg/model/service.go:629]: This is missing all the logic in the aggregate controller GetIstioServiceAccounts unless I am missing something. And if we don't need GetIstioServiceAccounts we should remove it.

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: pilot/pkg/model/service.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/model/service.go:629]: Ah.. That is a great catch. I just looked at kubecontroller and changed - AggregateController has some additional logic. Will fix it.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**